File: prokom3/akar.c
#include <stdio.h>
#include <math.h>

int main()
{
    int a,b,c,d,x1,x2;
    printf("Program Persamaan Kuadrat Ax^2 + Bx + C\n");
    printf("---\n");
    printf("Masukkan A: ");
    scanf("%d", &a);
    printf("Masukkan B: ");
    scanf("%d", &b);
    printf("Masukkan C: ");
    scanf("%d", &c);
    
    d=(b*b)-(4*a*c);
    
    if ( a == 0 )
    {
        printf("Bukan persamaan kuadrat\n");
    }
    else 
    {
    	if (d == 0)
    	{
        	printf("Akar kembar\n");
        	x1=-b/(2*a);
        	printf("x1=x2= %d", x1);
    	}
    	else if (d > 0 )
   	 	{
        	x1=(-b+sqrt(d))/(2*a);
        	x2=(-b-sqrt(d))/(2*a);
        	printf("x1= %d\n", x1);
        	printf("x2= %d", x2);
    	}
    	else
    	{
        	printf("Kedua akar imajiner");
    	}
    }
    
    printf("\n");
    return 0;
}

File: code/octave/akar.m
  % akar.m 
% program untuk mencari akar-akar persamaan kuadrat

close all; clear all; clc;
a=input('Masukkan nilai A= ');
b=input('Masukkan nilain B= ');
c=input('Masukkan nilai C= ');

% next: cek apakan nilai A > 0

d=(b^2)-(4*a*c);
if b > 0;
    x1=(-b+sqrt(d))/(2*a);
    x2=(-b-sqrt(d))/(2*a);
    disp(['x1= ' num2str(x1)])
    disp(['x2= ' num2str(x2)])
else
    disp('akar imajiner')
end